iOS Client/Store/MBStore.swift

```diff
@@ -96,35 +129,60 @@ class MBStore: NSObject {
         performDownload(clientFunction: client.getArticlesWithCompletion, persistentContainer: persistentContainer, deserializeFunc: MBArticle.deserialize, completion: completion)
     }
     
-    private func loadDevotions(completion: @escaping ([MBDevotion]?, Error?) -> Void) {
-        
-        let manager = FileManager.default
-        guard let url = manager.urls(for: .documentDirectory, in: .userDomainMask).first as URL? else {
-            completion(nil, MBDeserializationError.contextInsertionError(msg: "Could not create Devotions path"))
-            return
-        }
-        let path = url.path.appending("/devotions")
-        let pathUrl = URL(fileURLWithPath: path)
-        if !FileManager.default.fileExists(atPath: path) {
-            client.getDevotionsWithCompletion { data, error in
-                if error == nil, let data = data {
-                    self.save(data, pathUrl) { error in
-                        if error != nil {
-                            print("Could not save devotions data to disk")
+    func markDevotionAsRead(date: String, completion: @escaping (Error?) -> Void) {
+        read(fromPath: "readDevotionDates", [String].self) { dates, error in
+            do {
+                if var readDates = dates {
+                    if error != nil {
+                        print("could not mark devotion as read")
+                    } else if !readDates.contains { $0 == date } {
+                        readDates.append(date)
+                        let encoder = JSONEncoder()
+                        let (_, _, url) = try self.urlPackage(forPath: "/readDevotionDates")
+                        let data = try encoder.encode(readDates)
+                        self.save(data, url){ error in
+                            completion(error)
                         }
                     }
-                    self.parse(data, [MBDevotion].self, completion)
                 } else {
-                    completion(nil, error)
+                    throw StoreError.readError(msg: "couldn't read dates from documents")
                 }
+            } catch let error {
+                completion(error)
             }
-        } else if let data = FileManager.default.contents(atPath: path) {
-            parse(data, [MBDevotion].self, completion)
-        } else {
-            completion(nil, MBDeserializationError.fetchError(msg: "Could not fetch devotions"))
         }
     }
     
+    
+    private func read<T: Codable>(fromPath: String, _ resource: T.Type, completion: @escaping (T?, Error?) -> Void) {
+        do {
+            let (manager, path, _) = try urlPackage(forPath: "/\(fromPath)")
+            if !manager.fileExists(atPath: path), let file = Bundle.main.url(forResource: fromPath, withExtension: "json") {
+                    let data = try Data(contentsOf: file)
```

Comment:
I miss the notion you had before of moving the data from the external source to a new location inside our document directory. I think it's important to decouple the data source from our internal store